ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter-input/filter-location-input.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import React from 'react'
+import withListenTo from '../../backbone-container'
+const LocationView = require('../../location/index.js')
+const LocationOldModel = require('../../../component/location-old/location-old')
+const {
+  Direction,
+} = require('../../../component/location-new/utils/dms-utils.js')
+const ShapeUtils = require('../../../js/ShapeUtils.js')
+const CQLUtils = require('../../../js/CQLUtils.js')
+const wreqr = require('../../../js/wreqr.js')
+const store = require('../../../js/store.js')
+const wkx = require('wkx')
+import { deserialize } from '../../../component/location-old/location-serialization'
+
+const filterToLocationOldModel = filter => {
+  if (filter === '') return filter
+
+  if (typeof filter.geojson === 'object') {
+    return deserialize(filter.geojson)
+  }
+
+  // for backwards compatability with wkt
+  if (filter.value && typeof filter.value === 'string') {
```

Comment:
Is it feasible for none of the three if statements in this function to be executed?
Should we add a default return just in case?
if thats the case you might be able to refactor a bit to get rid of L30 and `return ''` if none of the other returns are hit

Reply:
It shouldn't be possible, because value either comes in as a previously created location model or a blank string. I was able to remove the `return ''` because the `deserialize` already handled that check here: https://github.com/codice/ddf/pull/5285/files#diff-4d9dce5d54ab93cb28c8979f2ce79b6fR102